### DISCUSSION TITLE
Development setup?

### DISCUSSION THREAD

**Discussion by phankhanhhung** [Budibase/budibase#10908] [Help] [answered]
Hi all,
Please help me the guide to deploy a development environment so that I can run, debug budibase?

Thank you very much.

**Comment by andz-bb** [comment, marked as answer]
hey @phankhanhhung 

thanks for your interest in contributing to Budibase!

Here are a few docs on the repo that should help you to get started with the development environment:

https://github.com/Budibase/budibase/blob/develop/docs/CONTRIBUTING.md
https://github.com/Budibase/budibase/blob/develop/docs/DEV-SETUP-MACOSX.md

Hope these help, if you run into any issues, let me know!

**Reply by phankhanhhung** [reply]
I have this error while run yarn setup:

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of '[git@github.com](mailto:git@github.com):Budibase/budibase-pro.git' into submodule path '/Users/hungphan/budibase2/packages/pro' failed
Failed to clone 'packages/pro' a second time, aborting
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

please help.

**Reply by phankhanhhung** [reply]
I removed git config submodule.recurse true && git submodule update in the package.json, hope it will help. :)
